webapp/src/js/components/Plot.js

```diff
@@ -32,7 +30,24 @@ let Plot = createReactClass({
     };
   },
 
+  componentWillMount() {
+    Promise.all([
+      import('plotly.js/dist/plotly-cartesian'),
+      //...more imports
+    ]).then(([Plotly]) => {
+      this.hasImported = true;
+      this.Plotly = Plotly;
+      this.PlotlyComponent = createPlotlyComponent(this.Plotly);
+      this.forceUpdate();
```

Comment:
Why not store plotly component in state?
`this.setState({plotlyComponent: createPlotlyComponent(Plotly)})
and then use in render?

Reply:
Good question; I don't know. What are the pros and cons?

Reply:
Yes, I think that will generally lead to cleaner code; trying it out now.